packages/cli/src/fulltext/FileIndex.ts

```diff
@@ -298,3 +304,20 @@ export async function filterFiles(
   }
   return result;
 }
+
+// Reads git attribute patterns from a .gitattributes file.
+export function getGitAttributes(directory: string) {
+  const gitAttributesPath = join(directory, '.gitattributes');
+  if (!existsSync(gitAttributesPath)) return [];
+
+  const gitAttributesContent = readFileSync(gitAttributesPath, 'utf-8');
+  const lines = gitAttributesContent.split('\n').filter(Boolean);
+
+  return lines.map((line) => {
+    const [pattern, ...attributes] = line.split(/\s+/);
+    return {
+      pattern,
+      binary: attributes.includes('binary'),
+    };
```

Comment:
If it's only pulling binary like that, why even bother returning other patterns? Perhaps it should just return a list of binary patterns instead. This will also simplify upstream code — you can just add those to excluded patterns.